apps/frontend/src/libs/components/protected-component/protected-component.tsx

```diff
@@ -0,0 +1,36 @@
+import { DataStatus, type PermissionKey } from "~/libs/enums/enums.js";
+import { checkHasPermission } from "~/libs/helpers/helpers.js";
+import { useAppSelector } from "~/libs/hooks/hooks.js";
+import { type ValueOf } from "~/libs/types/types.js";
+
+type Properties = {
+	children: React.ReactNode;
+	requiredPermissions: ValueOf<typeof PermissionKey>[];
+};
+
+const ProtectedComponent = ({
```

Comment:
Let's remove this wrapper component for two reasons:

1. We don't need to check for authenticatedUser loading, because in places where we are supposed to check for permissions, we should already have a user. If it's not the case, then the issue is somewhere else where we allowed the component to be rendered before we have a user. We need to show a loader for the whole app when user is loading.
2. We may handle missing permissions differently for components, not always we want to hide the component, but sometimes, we disable, or add additional tooltips.
 
Not to recalculate permissions everywhere using flatMap, we can add permissions to the auth slice

Reply:
Agree. Thanks for suggestion!